src/app/content/highlights/utils/getHighlightByIdFromHighlights.ts

```diff
@@ -0,0 +1,17 @@
+import { Highlight } from '@openstax/highlighter/dist/api';
+import { HighlightData } from '../types';
+
+const getHighlightByIdFromHighlights = (highlights: HighlightData[], id: string) => {
+  let foundHighlight: Highlight | undefined;
+
+  const oldHiglightIndex = highlights.findIndex(
+    (highlight) => highlight.id === id);
+
+  if (oldHiglightIndex < 0) { return foundHighlight; }
+
+  foundHighlight = {...highlights[oldHiglightIndex]};
+
+  return foundHighlight;
```

Comment:
this function seems excessively complicated. before the index was being fetched so that it could then be used to write back in the new highlight object, in this case there is no reason to findIndex you could use the regular find. you're also copying the object here which is undocumented and unnecessary